nano/scripts/nano/nano_window.js

```diff
@@ -1,25 +1,78 @@
 var NanoWindow = function () 
 {
+    var offsetX, offsetY;
+    var minWidth = 280;
+    var minHeight = 140;
+
     var setPos = function (x, y) {
         NanoUtility.winset("pos", x + "," + y);
     };
     var setSize = function (w, h) {
        NanoUtility.winset("size", w + "," + h);
     };
 
+    var obeyLimits = function () {
+        // Alignment (Thanks Goon (CC BY-NC-SA v3))
+        var prevX = window.screenLeft;
+        var prevY = window.screenTop;
+        //Put the window at top left
+        setPos(0, 0);
+        //Get any offsets still present
+        offsetX = window.screenLeft;
+        offsetY = window.screenTop;
+        // Clamp it to the viewport
+        var clampedX = prevX - offsetX;
+        var clampedY = prevY - offsetY;
+        clampedX = clampedX < offsetX ? 0 : clampedX;
+        clampedY = clampedY < offsetY ? 0 : clampedY;
+        // Put it back
+        setPos(clampedX, clampedY)
+        // Done with alignment
+
+        // Size constraints
+        var width = document.body.offsetWidth;
+        var height = Math.max(document.body.scrollHeight, document.body.offsetHeight,
+            document.documentElement.clientHeight, document.documentElement.scrollHeight, document.documentElement.offsetHeight);
+        var newWidth = Math.max(minWidth, width);
+        var newHeight = Math.max(minHeight, height);
+        setSize(newWidth, newHeight);
+    };
+
     var init = function () {
-        if(NanoStateManager.getData().config.user.fancy) {
-            fancyChrome();
-            attachButtons();
-            attachDragAndResize();
+        /* We want nanoUI windows to obey the limits regardless, 
+        and we're hijacking this library to do so, even if fancy mode is turned off.
+        To be fair, this doesn't stop the user from shrinking them past their intended limit when they're not in fancy mode, 
+        but it's a nice safeguard against shoddily-coded UIs. */
+        setupScroll();
+        obeyLimits();
+
+        if(!NanoStateManager.getData().config.user.fancy) {
+            return
         }
+
+        fancyChrome();
+        // ... but when fancy mode is turned on, we have to do it again to make sure
+        // that the window offsets are correct for the slightly different window.
+        obeyLimits();
+ 
+        attachButtons();
+        attachDragAndResize();
+    };
+
+    var setupScroll = function() {
+        $("#cornerWrapper").focus();
     };
+
     var fancyChrome = function() {
         NanoUtility.winset("titlebar", 0);
         NanoUtility.winset("can-resize", 0);
+        NanoUtility.winset("transparent-color", "#FF00E4")
```

Comment:
For anyone curious, this is what makes all the stupid magic transparency shit happen. BYOND supports this property on any `Main` elements, and it makes it so that any colors within it matching `#FF00E4` (bright pink) turn completely transparent.